contractcourt/contract_resolvers.go

```diff
@@ -449,27 +449,30 @@ func (h *htlcSuccessResolver) Resolve() (ContractResolver, error) {
 				return nil, err
 			}
 
-			// With out address obtained, we'll query for an
+			// With our address obtained, we'll query for an
 			// estimate to be confirmed at ease.
 			//
 			// TODO(roasbeef): signal up if fee would be too large
 			// to sweep singly, need to batch
-			feePerVSize, err := h.FeeEstimator.EstimateFeePerVSize(6)
+			feePerKw, err := h.FeeEstimator.EstimateFee(6)
 			if err != nil {
 				return nil, err
 			}
 
-			log.Debugf("%T(%x): using %v sat/vbyte to sweep htlc"+
+			log.Debugf("%T(%x): using %v sat/kw to sweep htlc"+
 				"incoming+remote htlc confirmed", h,
-				h.payHash[:], int64(feePerVSize))
+				h.payHash[:], int64(feePerKw))
 
 			// Using a weight estimator, we'll compute the total
 			// fee required, and from that the value we'll end up
 			// with.
-			totalVSize := (&lnwallet.TxWeightEstimator{}).
-				AddWitnessInput(lnwallet.OfferedHtlcSuccessWitnessSize).
-				AddP2WKHOutput().VSize()
-			totalFees := feePerVSize.FeeForVSize(int64(totalVSize))
+			var weightEstimator lnwallet.TxWeightEstimator
+			weightEstimator.AddWitnessInput(
+				lnwallet.OfferedHtlcSuccessWitnessSize,
+			)
+			weightEstimator.AddP2WKHOutput()
+			totalWeight := weightEstimator.Weight()
```

Comment:
The chaining previously used read pretty nicely IMO :)

Reply:
Fixed.